### ISSUE TITLE
NET6 Controller Action with Cancellation Token throws exception when auditing

### ISSUE THREAD

**Issue by wouter-b** [thepirat000/Audit.NET#536]
**Describe the bug**
When auditing a controller action with cancellation token an exception is thrown
```csharp
    [HttpGet("ok-with-cancellation-token")]
    public IActionResult GetOkWithCancellationToken(CancellationToken cancellationToken) => Ok();
```

```
System.NotSupportedException: Serialization and deserialization of 'System.IntPtr' instances are not supported. Path: $.Action.ActionParameters.WaitHandle.Handle.
 ---> System.NotSupportedException: Serialization and deserialization of 'System.IntPtr' instances are not supported.
   at System.Text.Json.Serialization.Converters.UnsupportedTypeConverter`1.Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWriteAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Converters.DictionaryDefaultConverter`3.OnWriteResume(Utf8JsonWriter writer, TDictionary value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonDictionaryConverter`3.OnTryWrite(Utf8JsonWriter writer, TDictionary dictionary, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   --- End of inner exception stack trace ---
   at System.Text.Json.ThrowHelper.ThrowNotSupportedException(WriteStack& state, NotSupportedException ex)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore[TValue](JsonConverter jsonConverter, Utf8JsonWriter writer, TValue& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteStreamAsync[TValue](Stream utf8Json, TValue value, JsonTypeInfo jsonTypeInfo, CancellationToken cancellationToken)
   at System.Text.Json.JsonSerializer.WriteStreamAsync[TValue](Stream utf8Json, TValue value, JsonTypeInfo jsonTypeInfo, CancellationToken cancellationToken)
   at System.Text.Json.JsonSerializer.WriteStreamAsync[TValue](Stream utf8Json, TValue value, JsonTypeInfo jsonTypeInfo, CancellationToken cancellationToken)
   at Audit.Core.JsonAdapter.SerializeAsync(Stream stream, Object value)
   at Audit.Core.Providers.FileDataProvider.SaveFileAsync(String fullPath, AuditEvent auditEvent)
   at Audit.Core.Providers.FileDataProvider.InsertEventAsync(AuditEvent auditEvent)
   at Audit.Core.AuditScope.SaveEventAsync(Boolean forceInsert)
   at Audit.Core.AuditScope.EndAsync()
   at Audit.Core.AuditScope.DisposeAsync()
   at Audit.WebApi.AuditApiAdapter.AfterExecutedAsync(ActionExecutedContext context, Boolean includeModelState, Boolean includeResponseBody, Boolean includeResponseHeaders)
   at Audit.WebApi.AuditApiGlobalFilter.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**To Reproduce**
A NET6 minimal api project
```csharp
using Audit.WebApi;
using Microsoft.AspNetCore.Mvc;

Audit.Core.Configuration.Setup().UseFileLogProvider(cfg => cfg.FilenamePrefix("auditnet_"));

var builder = WebApplication.CreateBuilder();
builder.Services.AddSwaggerGen();
builder.Services.AddControllers(cfg => cfg.AddAuditFilter(select => select.LogAllActions()));

var app = builder.Build();
app.UseSwagger();
app.UseSwaggerUI();
app.MapDefaultControllerRoute();
app.Run();

public class Controller : ControllerBase
{
    [HttpGet("ok-with-cancellation-token")]
    public IActionResult GetOkWithCancellationToken(CancellationToken cancellationToken) => Ok();

    [HttpGet("ok-without-cancellation-token")]
    public IActionResult GetOkWithOutCancellationToken() => Ok();
}
```

**Expected behavior**
No exception and the controller action gets audited

**Libraries (specify the Audit.NET extensions being used including version):**
```xml
    <PackageReference Include="Audit.WebApi.Core" Version="20.0.0" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0" />
```

**Target .NET framework:**
For example:
 - .NET 6

**Additional context**
This only occurs with the System.Text.Json serializer, with the JSON.Net (NewtonSoft) serializer no exception is thrown and the controller action gets audited.

**workaround**
```csharp
Audit.Core.Configuration.JsonAdapter = new JsonNewtonsoftAdapter()
````

**suggested solution**
Clearly this is an issue with the System.Text.Json serializer. But there's no need to serialize the Cancellation Token.
Excluding the cancellation token from the parameters to serialize will solve this issue.

**Comment by thepirat000** [comment]
I think the fix will be to filter out the parameters to include in the audit, to only consider those that use input from the request.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.bindingsource.isfromrequest?view=aspnetcore-6.0

I'll work on this fix as soon as I can

**Comment by thepirat000** [comment]
This eas fixed on version 20.0.1, please upgrade your references and re-test

**Comment by wouter-b** [comment]
Thank you! My issue has been fixed.
